### PR TITLE
modernise .travis.yml

### PR DESCRIPTION
**Pull request by John-Colvin**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,29 @@
+sudo: false
```
Comment (John-Colvin) [inline .travis.yml]: so as to use the new container-based infrastructure

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
+
 language: d
 
-#compiler:
-#  - dmd
+os:
+ - linux
+ - osx
```
Comment (John-Colvin) [inline .travis.yml]: The OS X testers aren't enabled by default, but if you send a quick email to support@travis-ci.com they will be (hopefully) happy to enable them for you. I have had no problem getting them for my projects.

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
+ - dmd-2.068.0
+ - dmd-2.067.1
+ - dmd-2.066.1
+ - ldc-0.15.1
+ - gdc
+ 
+matrix:
+  allow_failures:
+   - os: osx
+     d: gdc
```
Comment (John-Colvin) [inline .travis.yml]: gdc isn't working on the OS X testers at the moment

File: .travis.yml [hunk 4]
```diff
@@ -1,7 +1,29 @@
...
+   - env: ARCH=x86
+     d: gdc
```
Comment (John-Colvin) [inline .travis.yml]: gdc 32bit isn't working on the linux tester at the moment

File: .travis.yml [hunk 5]
```diff
@@ -9,29 +31,12 @@ notifications:
 env:
   global:
     - secure: "wnthDxOwv7qaVfJQfwYQLFo4DDBgfT4HoLntBsfmYjwmSVzCuXPktCK3vOpimb5twyiVNkt1I/82x8lyYpiczKqPcNZqd8p91mInt5eKIf8Gacqb3a1FwGyk9bwHSZdYpQiblY/wMVb0+WJTuhlVf1NI5xXMWCf1l88+1q6E6jo="
```
Comment (John-Colvin) [inline .travis.yml]: this isn't the most secure thing to do, no? See http://docs.travis-ci.com/user/environment-variables/

File: .travis.yml [hunk 6]
```diff
@@ -9,29 +31,17 @@ notifications:
...
 
 script:
-  - dub test
-  - dub test --arch=x86
+  - if [ $DC == ldc2 ] && [ $ARCH == x86_64 ];
+    then
+      dub test;
+    else
+      dub test --arch=$ARCH;
+    fi
```
Comment (John-Colvin) [inline .travis.yml]: A workaround for a bug in dub 0.9.23 that will hopefully go away in 0.9.24

**Comment by John-Colvin** [comment]
The test failure here is not a problem with this pull, it's just showing that LuaD doesn't work with dmd 2.067.1 or 2.068.0

File: .travis.yml (hunk 5 above)
Comment (JakobOvrum) [inline .travis.yml]: On that same page it seems to be the correct way to do it.

Comment (John-Colvin) [inline .travis.yml]: yes, sorry, my mistake

**Comment by JakobOvrum** [comment]
Thanks a lot! Yeah, #93 is fairly critical.

**Comment by John-Colvin** [comment]
No problem, I'm hoping I will be able to use LuaD to get D interoperability with http://torch.ch and more generally to compliment the interfaces with python and R that already exist (the R one is newer, pyd has some new wrappers that I'm working on) so that DlangScience (https://dlangscience.github.io, https://dlangscience.github.io) can make strong claims about interoperability with other popular scientific programming eco-systems and their comprehensive libraries.

**Pull request merged**